actors/runtime/src/main/java/cloud/orbit/actors/runtime/Hosting.java

```diff
@@ -254,10 +260,11 @@ private void onClusterViewChanged(final Collection<NodeAddress> nodes)
         consistentHashNodeTree = newHashes;
         updateServerNodes();
 
+        Collection<NodeAddress> values = localAddressCache.asMap().values();
         for (NodeInfo info : oldNodes.values())
         {
             // clear local cache
-            localAddressCache.values().remove(info.address);
+            values.remove(info.address);
```

Comment:
Sorry. There was a comment about doing an iteration to get the keys and then use remove all. I missed the outer loop and misread your stream as that suggestion in code. Your removeAll is best.